### DISCUSSION TITLE
Problem to extend img tags with additional props

### DISCUSSION THREAD

**Discussion by Dexus** [solidjs/solid#2317] [General]
I have the problem that I use a script :  https://github.com/DeineAgenturUG/ionic-image-cache (changes not yet pushed)

that are cache images to the indexeddb but use a `url` insted of an `src` while templateing. But the `url` will never rendered to the browser. On nested pages.

Pages, that are direct component of the route have also a `<img url="https...." />` prop, and they get rendered right, but once i have in my same probs with in nested children it will removed.

I have a declare added to make it work and don't show me anything red...

```
declare module "solid-js" {
    namespace JSX {
        interface ImgHTMLAttributes<T> {
            is?: string;
            expire?: string;
            url?: string;
            lazy?: boolean;
        }
    }
}
```

What did I miss to make it working in all components?

**Comment by trusktr** [comment]
Can you show what the code looks like that is not working?

Might be a problem of Solid setting attributes vs JS properties, but I'm not sure. If so, then the ideas here,

- https://github.com/solidjs/solid/discussions/2322

may help fix inconsistencies.

Need to see your code to understand exactly what you mean.
